From 4e26d97a63c56147cfaf9d1a81c1a198e4ed6788 Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Wed, 18 Jan 2012 10:38:16 +0100
Subject: [PATCH 22/52] vhost_net.c: v2 Fix build failure introduced by
 0bfcd599e3f5c5679cc7d0165a0a1822e2f60de2

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1326883126-22053-23-git-send-email-armbru@redhat.com>
Patchwork-id: 36590
O-Subject: [RHEL-6.3 PATCH qemu-kvm 22/52] vhost_net.c: v2 Fix build failure introduced by 0bfcd599e3f5c5679cc7d0165a0a1822e2f60de2
Bugzilla: 758194
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>

From: Jes Sorensen <Jes.Sorensen@redhat.com>

Fix build failure introduced by 0bfcd599e3f5c5679cc7d0165a0a1822e2f60de2

The format statement expects unsigned long on x86_64, but receives
unsigned long long, so gcc exits with an error.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Blue Swirl <blauwirbel@gmail.com>
(cherry picked from commit 29f91781bd300b228d9c38b9f6c1e0b6a38f25fb)
Signed-off-by: Markus Armbruster <armbru@redhat.com>
---
 hw/vhost_net.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/vhost_net.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hw/vhost_net.c b/hw/vhost_net.c
index b099efd..cb666d0 100644
--- a/hw/vhost_net.c
+++ b/hw/vhost_net.c
@@ -116,7 +116,7 @@ struct vhost_net *vhost_net_init(VLANClientState *backend, int devfd,
     }
     if (~net->dev.features & net->dev.backend_features) {
         fprintf(stderr, "vhost lacks feature mask %" PRIu64 " for backend\n",
-                ~net->dev.features & net->dev.backend_features);
+                (uint64_t)(~net->dev.features & net->dev.backend_features));
         vhost_dev_cleanup(&net->dev);
         goto fail;
     }
-- 
1.7.7.5

